ocaml/jane/doc/extensions/unboxed-types/index.md

```diff
@@ -284,4 +284,144 @@ Here's the list of primitives that currently support `[@layout_poly]`:
 * `%array_safe_get`
 * `%array_safe_set`
 * `%array_unsafe_get`
-* `%array_unsafe_set`
\ No newline at end of file
+* `%array_unsafe_set`
+
+# Using unboxed types in structures
+
+Unboxed types can usually be put in structures, though there are some restrictions.
+
+These structures support unboxed type fields, but have some restrictions on field
+orders:
+  * Records
+  * Constructors
+  
+Unboxed numbers can't be put in these structures:
+  * Constructors with inline record fields
+  * Exceptions
+  * Extensible variant constructors
+  * Top-level fields of modules
+  * Tuples
+  
+There aren't fundamental issues with the structures that lack support. They will
+just take some work to implement.
+
+Here's an example of a record with some unboxed fields. We call such a record 
+a "mixed record".
+
+```ocaml
+type t =
+  { str : string;
+    i : int;
+    f : float#;
+  }
+```
+
+## Restrictions on ordering of fields
+
+The below is written about record fields but equally applies to constructor
+arguments.
+
+Suppose a record contains any unboxed field `fld` whose layout is not `value` or a
+combination of `value` layouts. Then, the following restriction applies: All
```

Comment:
Do combinations of `value` exist? Everything that's not unboxed should just be `value` right?

Reply:
I don't want to need to reword this when we have unboxed tuples (imminent) or unboxed sums (someday).

Reply:
I'll push the technicality into a footnote. I brainstormed better phrasings with Chris and came up empty.